batchflow/models/validator/README.md

```diff
@@ -0,0 +1,224 @@
+There are two main reasons to wrap your model using classes from `Validator` submodule:
+
+- make the structure of the model lifecycle clear,
+- provide API to automized model training and validation.
+
+All you need is:
+
+* define `model_api.py` file with model inherited from `ModelAPI`,
+* configure it by `validator.yaml` file in the same folder,
+* execute the following code from the same folder:
+
+```python
+from batchflow.models.validator import Validator
+
+val = Validator()
+val.start()
+print(val.metrics)
+```
+
+## Basic example
+
+### model_api.py
+
+```python
+import ...
+
+class PascalValidator(ModelAPI):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_config = ...
+        self.ds = PascalSegmentation(bar=True)
+
+    def train_loader(self, **kwargs):
+        return self.ds.train
+
+    def test_loader(self, **kwargs):
+        return self.ds.test
+
+    def train(self, ds):
+        train_ppl = ... << ds
+        train_ppl.run()
+        return train_ppl.get_model_by_name('model')
+
+    def inference(self, ds, model):
+        test_ppl = ... << ds
+        test_ppl.run()
+        targets = np.array(test_ppl.v('targets'))
+        predictions = np.array(test_ppl.v('predictions'))
+        return targets, predictions
+```
+
+### validator.yaml
+
+```yaml
+- Pascal Segmentation:
+  - PascalValidator:
+      class: PascalValidator
+      train:
+      test:
+        metrics:
+        - classification:
+            evaluate:
+              - accuracy
+```
+
+### `val.metrics`
+
+```python
+{'Pascal Segmentation':
+       {'PascalValidator':
+              {'classification': {'accuracy': 0.0092724609375}
+       }
+}
+```
+
+## **model_api**
+
+`ModelAPI` child-class can implement the following methods:
+
+### `train_loader(self, path=None, **kwargs)`
+
+`path` and `kwargs` are from config `<task_name>/<model_name>/<train>`. In the example above we have empty value for `train` key, therefore `path=None` and `kwargs={}`. Let's define config:
+
+```yaml
+- Pascal Segmentation:
+  - PascalValidator:
+      class: PascalValidator
+      train:
+        dataset: /path/to/dataset
+```
+
+In that case `path='/path/to/dataset'`. You also can define multiple parameters of `train_loader`:
+
+```yaml
+- Pascal Segmentation:
+  - PascalValidator:
+      class: PascalValidator
+      train:
+         dataset:
+           path: /path/to/dataset
+           format: 'png'
+```
+
+Now `path='/path/to/dataset', kwargs={format: 'png'}`.
+The output of the function will be used as `train_dataset` argument of `train` method. By default, it returns `path`.
+
+### `test_loader(self, path=None, **kwargs)`
+
+The same as `train_loader` but for `test`.
+
+### `load_model(self, path=None, **kwargs)`
+
+Loader for pretrained model. Let's make example above more complex:
+
+```yaml
+- Pascal Segmentation:
+  - PascalValidator:
```

Comment:
The second level os nesting may be omitted as well as the first. Let's make things simpler. Each yaml file for one model. One can have as many yaml files as he needs.

Reply:
`---` can also be used to separate different validation definitions within one yaml file

Reply:
I like idea with multiple files.